docs/Grammar.md

```diff
@@ -155,3 +158,25 @@ untrusted_suffixes = "transition_using_threads" | "propagate_errno"
 
 
 ```
+
+### C-style directives
+
+```c
+
+ifdef_statement:
+    "#ifdef" string statement* [#else statement*] "#endif"
```

Comment:
nit: As implemented currently it is `"#" "ifdef"'`. That is spaces are supported between # and ifdef.

Reply:
Note. Unlike C preprocessor, we support other stuff in the same line...

```c
#ifdef P public void ecall(); // This is valid currently.
```

Reply:
Updated.